Minesweeper.java

```diff
@@ -81,24 +105,57 @@ private void run() {
             setVisible(true);
         }
 
+        public void revealEmptyCells(int row, int col) {
+            // Check bounds to avoid array out-of-bounds exceptions
+            if (row < 0 || col < 0 || row >= Game.cells.length || col >= Game.cells[0].length) {
+                return;
+            }
+            
+            Cell cell = Game.cells[row][col];
+            
+            // Check if the cell is already revealed or contains a mine
+            if (cell.isRevealed || cell.isBomb) {
+                return;
+            }
+            
+            // Reveal the cell
+            Game.cells[row][col].reveal();
+            
+
+            // If this cell has no adjacent mines, recursively reveal its neighbors
+            if (cell.getNeighboringBombs(Game.cells[row][col]) == 0) {
+                // Define relative positions of neighboring cells
+                int[] dr = {-1, -1, -1, 0, 1, 1, 1, 0};
+                int[] dc = {-1, 0, 1, 1, 1, 0, -1, -1};
+                
+                for (int i = 0; i < 8; i++) {
+                    int newRow = row + dr[i];
+                    int newCol = col + dc[i];
+                    revealEmptyCells(newRow, newCol); // Recursively check neighbors
+                }
+            }
+        }
+        
+
+
         private void populateBombs() {
             Random random = new Random();
             int remainingBombs = this.bombAmount;
 
             while (remainingBombs > 0) {
-                int randomColumn = random.nextInt(this.gridSize);
-                int randomRow = random.nextInt(this.gridSize);
+                int randomColumn = random.nextInt(Game.gridSize);
+                int randomRow = random.nextInt(Game.gridSize);
                 
-                if (this.cells[randomRow][randomColumn].isBomb) {
+                if (Game.cells[randomRow][randomColumn].isBomb) {
                     continue;
                 }
 
-                this.cells[randomRow][randomColumn].makeBomb();
+                Game.cells[randomRow][randomColumn].makeBomb();
                 remainingBombs--;
             }
         }
 
-        private void computeNeighboringBombs(Cell cell) {
+        private static int computeNeighboringBombs(Cell cell) {
```

Comment:
this change isn't needed if we use `this` instead of `Game`, which we can if we get rid of the `getNeighboringBombs` method from `Cell`.
```suggestion
        private int computeNeighboringBombs(Cell cell) {
```